### PR TITLE
python312Packages.fs: clean-up

### PR DESCRIPTION
**Pull request by fabaff**
- enable tests

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Pandapip1** [comment]
This should probably target staging or staging-next?

**Comment by FliegendeWurst** [comment]
@ofborg build python312Packages.fs

**Review by Pandapip1 — APPROVED** [review]
Diff LGTM; borgy is happy

File: pkgs/development/python-modules/fs/default.nix [hunk 1]
```diff
@@ -30,13 +29,10 @@ buildPythonPackage rec {
 
   build-system = [ setuptools ];
 
-  buildInputs = [ glibcLocales ];
-
   dependencies = [
     six
     appdirs
     pytz
-    setuptools
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/fs/default.nix:39]: This change breaks some downstream packages; for example attempting to build `nomnatong` on current `master` (9e09ce2d47c7700b877becef99ce7e27f15604dc) results in

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/6kb205kz8m53bbgcs1xz5890zj4b6bad-source
source root is source/src
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Traceback (most recent call last):
  File "/nix/store/csrwfkw01nys8knp67xhxhdg1n11jlwc-python3.12-afdko-4.0.1/bin/.makeotf-wrapped", line 6, in <module>
    from afdko.makeotf import main
  File "/nix/store/csrwfkw01nys8knp67xhxhdg1n11jlwc-python3.12-afdko-4.0.1/lib/python3.12/site-packages/afdko/makeotf.py", line 13, in <module>
    from afdko import convertfonttocid, fdkutils, ufotools
  File "/nix/store/csrwfkw01nys8knp67xhxhdg1n11jlwc-python3.12-afdko-4.0.1/lib/python3.12/site-packages/afdko/ufotools.py", line 13, in <module>
    from fontTools.ufoLib import UFOReader
  File "/nix/store/1pjjzg67xrnk5m6b053k23c0n0m3s03k-python3.12-fonttools-4.54.1/lib/python3.12/site-packages/fontTools/ufoLib/__init__.py", line 42, in <module>
    import fs
  File "/nix/store/xbk7nbi14ihmnjxdzphjxpifs3104j3d-python3.12-fs-2.4.16/lib/python3.12/site-packages/fs/__init__.py", line 4, in <module>
    __import__("pkg_resources").declare_namespace(__name__)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'pkg_resources'
```

Comment (Pandapip1) [reply pkgs/development/python-modules/fs/default.nix:39]: That's a bug with those packages. `build-system = [ setuptools ];` should be added to those packages.

Comment (eclairevoyant) [reply pkgs/development/python-modules/fs/default.nix:39]: The stacktrace literally points to this package...

Comment (Pandapip1) [reply pkgs/development/python-modules/fs/default.nix:39]: Fair enough. I'll submit a patch.

**Pull request merged**